pkg/controller/elasticsearch/remotecluster/remoteca/controller.go

```diff
@@ -0,0 +1,474 @@
+	)
+
+	// Check if local CA exists
+	localCA := &corev1.Secret{}
+	if err := r.Client.Get(transport.PublicCertsSecretRef(localClusterKey), localCA); err != nil {
+		if !errors.IsNotFound(err) {
+			return results.WithError(err)
+		}
+		results.WithResult(defaultRequeue)
+	}
+
+	if len(localCA.Data[certificates.CAFileName]) == 0 {
+		log.Info(
+			"Cannot find local CA cert",
+			"local_namespace", localClusterKey.Namespace,
+			"local_name", localClusterKey.Namespace,
+		)
+		r.recorder.Event(local, v1.EventTypeWarning, EventReasonLocalCaCertNotFound, caCertMissingError("local", localClusterKey))
+		// CA secrets are watched, we don't need to requeue.
+		// If CA is created later it will trigger a new reconciliation.
+		return nil
+	}
+
+	// Check if remote CA exists
+	remoteCA := &corev1.Secret{}
+	if err := r.Client.Get(transport.PublicCertsSecretRef(remoteClusterKey), remoteCA); err != nil {
+		if !errors.IsNotFound(err) {
+			return results.WithError(err)
+		}
+		results.WithResult(defaultRequeue)
+	}
+
+	if len(remoteCA.Data[certificates.CAFileName]) == 0 {
+		log.Info("Cannot find remote CA cert",
+			"remote_namespace", remote.Namespace,
+			"remote_name", remote.Name,
+		)
+		r.recorder.Event(local, corev1.EventTypeWarning, EventReasonRemoteCaCertMissing, caCertMissingError("remote", remoteClusterKey))
+		return nil
+	}
+
+	// Create local relationship
+	if err := reconcileRemoteCA(ctx, r.Client, local, remoteClusterKey, remoteCA.Data[certificates.CAFileName]); err != nil {
+		return results.WithError(err)
+	}
+
+	// Create remote relationship
+	if err := reconcileRemoteCA(ctx, r.Client, remote, localClusterKey, localCA.Data[certificates.CAFileName]); err != nil {
+		return results.WithError(err)
+	}
+
+	return nil
+}
+
+// reconcileRemoteCA copies certificates authorities across 2 clusters
+func reconcileRemoteCA(
+	ctx context.Context,
+	c k8s.Client,
+	owner *esv1.Elasticsearch,
+	remote types.NamespacedName,
+	remoteCA []byte,
+) error {
+	span, _ := apm.StartSpan(ctx, "reconcile_remote_ca", tracing.SpanTypeApp)
+	defer span.End()
+
+	// Define the desired remote CA object, it lives in the remote namespace.
+	expected := corev1.Secret{
+		ObjectMeta: remoteCAObjectMeta(remoteCASecretName(owner.Name, remote), owner, remote),
+		Data: map[string][]byte{
+			certificates.CAFileName: remoteCA,
+		},
+	}
+
+	var reconciled corev1.Secret
+	return reconciler.ReconcileResource(reconciler.Params{
+		Client:     c,
+		Scheme:     scheme.Scheme,
+		Owner:      owner,
+		Expected:   &expected,
+		Reconciled: &reconciled,
+		NeedsUpdate: func() bool {
+			return !maps.IsSubset(expected.Labels, reconciled.Labels) || !reflect.DeepEqual(expected.Data, reconciled.Data)
+		},
+		UpdateReconciled: func() {
+			reconciled.Labels = maps.Merge(reconciled.Labels, expected.Labels)
+			reconciled.Data = expected.Data
+		},
+	})
+}
+
+func caCertMissingError(location string, cluster types.NamespacedName) string {
+	return fmt.Sprintf(
+		CaCertMissingError,
```

Comment:
I think I would be fine with hardcoding the constant here to make the function easier to read